### PR TITLE
#169 emit an event to update action in excal url

### PR DESCRIPTION
**Pull request by asmaNode**

### CODE REVIEW

**Review by rezk2ll — APPROVED** [review]

File: src/linagora.esn.calendar/app/event-view/external-user/event-view-external-user.controller.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ const _ = require('lodash');
     }
 
     function changeParticipation(partstat) {
+      $rootScope.$emit('UPDATE_ACTION_EXCAL', self.linksMapping[partstat]);
       self.userAsAttendee.partstat = partstat;
```
Comment (huy-ta) [inline src/linagora.esn.calendar/app/event-view/external-user/event-view-external-user.controller.js:29]:
```suggestion
      $rootScope.$emit('UPDATE_ACTION_EXCAL', self.linksMapping[partstat]);

      self.userAsAttendee.partstat = partstat;
```

And you should have a constant for `UPDATE_ACTION_EXCAL`.

**Review by huy-ta — COMMENTED** [review]
Now you need to write tests for the changes.

File: src/linagora.esn.calendar/app/event-view/external-user/event-view-external-user.controller.js [hunk 2]
```diff
@@ -25,6 +25,7 @@ const _ = require('lodash');
     }
 
     function changeParticipation(partstat) {
+      $rootScope.$emit(CAL_EVENTS.UPDATE_ACTION_EXCAL, self.linksMapping[partstat]);
```
Comment (tuanlc) [inline src/linagora.esn.calendar/app/event-view/external-user/event-view-external-user.controller.js:28]: `calendar` and `excal` are 2 different SPAs, why there is a angular event beween them?

Comment (asmaNode) [reply src/linagora.esn.calendar/app/event-view/external-user/event-view-external-user.controller.js:28]: excal component use calendar component to display event detail (eventCalView) , 
the event is fired in calendar component (the parent) for that we must emit event on it ,
 after that the child component (excal) keep listening to upcoming event t

Comment (tuanlc) [reply src/linagora.esn.calendar/app/event-view/external-user/event-view-external-user.controller.js:28]: I mean if the logic is used by only `excal` SPA, then it is not `common` libs anymore.
Furthermore, why external user view code is maintained in `calendar` codebase?
So, I suggest we move this template and her logic to `excal` module. WDY?

Comment (asmaNode) [reply src/linagora.esn.calendar/app/event-view/external-user/event-view-external-user.controller.js:28]: i think is not the right moment to refactor this , eventCalView is used in inbox calendar (to send invitation)

Comment (tuanlc) [reply src/linagora.esn.calendar/app/event-view/external-user/event-view-external-user.controller.js:28]: As you wish. However, a ticket should be created to refactor it.

Comment (asmaNode) [reply src/linagora.esn.calendar/app/event-view/external-user/event-view-external-user.controller.js:28]: ok wil discuss with @renaudboyer  @huy-ta  about it

**Review by rezk2ll — APPROVED** [review]

**Review by huy-ta — APPROVED** [review]
Yes I think you should create an issue to refactor the code later.

**Review by renaudboyer — APPROVED** [review]

**Pull request merged**